.travis.yml

```diff
@@ -1,44 +1,67 @@
 language: cpp
 
-cache: ccache
 before_install:
-  - |
-      if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
-        git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '(\.rst$)|(^doc/)' || {
-          echo "Only docs were updated, stopping build process."
-          exit
-        }
-      fi
+- |-
+    case $TRAVIS_OS_NAME in
+      windows)
+        [[ ! -f C:/tools/msys64/msys2_shell.cmd ]] && rm -rf C:/tools/msys64
+        choco upgrade --no-progress -y msys2
+        export msys2='cmd //C RefreshEnv.cmd '
+        export msys2+='& set MSYS=winsymlinks:nativestrict '
+        export msys2+='& C:\\tools\\msys64\\msys2_shell.cmd -defterm -no-start'
+        export mingw64="$msys2 -mingw64 -full-path -here -c \$\* --"
+        export msys2+=" -msys2 -c \$\* --"
+        $msys2 pacman --sync --noconfirm --needed mingw-w64-x86_64-toolchain autoconf libtool automake make autoconf-archive pkg-config
```

Comment:
If we're using MSVC now, I _think_ we don't need `mingw-w64-x86_64-toolchain` here.